libraries/core_libs/consensus/include/vote_manager/vote_manager.hpp

```diff
@@ -54,10 +54,10 @@ class NextVotesForPreviousRound {
 
   bool enough_votes_for_null_block_hash_;
   blk_hash_t voted_value_;  // For value is not null block hash
-  size_t next_votes_size_;
   // <voted PBFT block hash, next votes list that have exactly 2t+1 votes voted at the PBFT block hash>
   // only save votes == 2t+1 voted at same value in map and set
   std::unordered_map<blk_hash_t, std::vector<std::shared_ptr<Vote>>> next_votes_;
+  std::unordered_map<blk_hash_t, uint64_t> next_votes_weight_;
```

Comment:
Since in current design, we save exactly 2t+1 next votes for previous round. In PBFT next voting state, node sometimes will votes on 2 values, NULL HASH and a voted value.
So in syncing next votes for previous round, we will always check if there are 2t+1 votes or 2 * (2t+1) votes.
The next_votes_weight_ will break that rule. Otherwise will cause lots of useless next votes syncing